src/common/SurgePatch.cpp

```diff
@@ -1459,7 +1461,7 @@ void SurgePatch::load_xml(const void* data, int datasize, bool is_preset)
    // ensure that filtersubtype is a valid value
    for (auto& sc : scene)
    {
-      for (int u = 0; u < 2; u++)
+      for (int u = 0; u < n_filterunits_per_scene; u++)
       {
          sc.filterunit[0].subtype.val.i =
```

Comment:
I think there's an old bug here that the [0] should be the [u] don't you agree? If so we should open a separate issue to investigate this line (your change is fine here).

Reply:
That definitely looks like a legit bug to me, because that inner `u` loop doesn't do anything other than repeat the same thing 2x...